src/Storages/ObjectStorage/DataLakes/DeltaLake/TableSnapshot.cpp

```diff
@@ -227,60 +227,63 @@ class TableSnapshot::Iterator final : public DB::IObjectIterator
         /// but instead in data files paths directory names.
         /// So we extract these values here and put into `partitions_info`.
         DB::ObjectInfoWithPartitionColumns::PartitionColumnsInfo partitions_info;
-        for (const auto & partition_column : context->partition_columns)
+        if (partition_map)
```

Comment:
Here I only added `if (partition_map)` check, other diff is from the scope change. And another change is check for `stat` in log message below in line 286.